tests/e2e/stablehlo_models/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-if(NOT IREE_INPUT_STABLEHLO)
+if(NOT IREE_INPUT_STABLEHLO AND NOT IREE_HOST_BIN_DIR)
```

Comment:
Bah. This re-enables the tests:

https://github.com/openxla/iree/actions/runs/7935247478/job/21668457807?pr=16440#step:8:549

```
******** Running run-module CTest ********
Internal ctest changing into directory: /work/build-linux-x86_64
Test project /work/build-linux-x86_64
    Start  22: iree/tools/test/iree_run_module_correctness_test
1/5 Test  #22: iree/tools/test/iree_run_module_correctness_test ......................   Passed    0.00 sec
    Start 413: iree/tests/e2e/stablehlo_models/mobilenet_v1_fp32_correctness_test
2/5 Test #413: iree/tests/e2e/stablehlo_models/mobilenet_v1_fp32_correctness_test ....   Passed    0.05 sec
    Start 414: iree/tests/e2e/stablehlo_models/efficientnet_int8_correctness_test
3/5 Test #414: iree/tests/e2e/stablehlo_models/efficientnet_int8_correctness_test ....   Passed    0.03 sec
    Start 415: iree/tests/e2e/stablehlo_models/deeplab_v3_fp32_correctness_test
4/5 Test #415: iree/tests/e2e/stablehlo_models/deeplab_v3_fp32_correctness_test ......   Passed    0.05 sec
    Start 416: iree/tests/e2e/stablehlo_models/person_detect_int8_correctness_test
5/5 Test #416: iree/tests/e2e/stablehlo_models/person_detect_int8_correctness_test ...   Passed    0.00 sec

100% tests passed, 0 tests failed out of 5

Label Time Summary:
driver=local-sync                  =   0.14 sec*proc (5 tests)
iree/tests/e2e/stablehlo_models    =   0.14 sec*proc (4 tests)
iree/tools/test                    =   0.00 sec*proc (1 test)
test-type=run-module-test          =   0.14 sec*proc (5 tests)

Total Test time (real) =   0.19 sec
```

That's still not worth all this CI overhead. Might be able to run the tests as part of another job.